File: dkplayer-sample/src/main/java/com/dueeeke/dkplayer/util/ProgressManagerImpl.java
package com.dueeeke.dkplayer.util;

import android.support.v4.util.LruCache;
import android.text.TextUtils;

import com.dueeeke.videoplayer.player.ProgressManager;

public class ProgressManagerImpl extends ProgressManager {

    //保存100条记录
    private static LruCache<Integer, Long> mCache = new LruCache<>(100);

    @Override
    public void saveProgress(String url, long progress) {
        if (TextUtils.isEmpty(url)) return;
        if (progress == 0) {
            clearSavedProgressByUrl(url);
            return;
        }
        mCache.put(url.hashCode(), progress);
    }

    @Override
    public long getSavedProgress(String url) {
        if (TextUtils.isEmpty(url)) return 0;
        Long pro = mCache.get(url.hashCode());
        if (pro == null) return 0;
        return pro;
    }

    public void clearAllSavedProgress() {
        mCache.evictAll();
    }

    public void clearSavedProgressByUrl(String url) {
        mCache.remove(url.hashCode());
    }
}

File: dkplayer-ui/src/main/java/xyz/doikki/videocontroller/component/ProgressManagerImpl.kt
package xyz.doikki.videocontroller.component

import android.util.LruCache
import xyz.doikki.videoplayer.util.orDefault

class ProgressManagerImpl : ProgressManager {

    //保存100条记录
    private val mCache = LruCache<Int, Long?>(100)

    private fun generateKey(url: String): Int {
        return url.hashCode()
    }

    override fun saveProgress(url: String, progress: Long) {
        if (url.isEmpty())
            return
        if (progress == 0L) {
            clear(url)
            return
        }
        mCache.put(generateKey(url), progress)
    }

    override fun getSavedProgress(url: String): Long {
        return if (url.isEmpty()) 0 else mCache[generateKey(url)].orDefault()
    }

    override fun clear(url: String) {
        mCache.remove(generateKey(url))
    }

    override fun clearAll() {
        mCache.evictAll()
    }


}